src/main/java/com/nurkiewicz/lazyseq/LazySeqStream.java

```diff
@@ -240,11 +241,34 @@ public DoubleStream flatMapToDouble(Function<? super E, ? extends DoubleStream>
 
 	@Override
 	public Stream<E> onClose(Runnable closeHandler) {
-		throw new UnsupportedOperationException("Not yet implemented: onClose");
+		Runnable previousCloseAction = closeAction;
+		closeAction = previousCloseAction == null ? closeHandler : () -> {
```

Comment:
Copies handling from `Streams.composeWithExceptions` used in `AbstractPipeline`, which is unfortunately internal API.